cms/djangoapps/contentstore/views/course.py

```diff
@@ -277,6 +281,22 @@ def course_handler(request, course_key_string=None):
         raise Http404
 
 
+@login_required
+def course_archive_handler(request, course_key_string=None):
+    if request.method == 'GET':
+        if course_key_string:
+            course_key = CourseKey.from_string(course_key_string)
+            course_details = CourseDetails.fetch(course_key)
+            end_date_setting = {
+                'start_date': (datetime.today() - timedelta(days=2)).replace(tzinfo=utc),
+                'end_date': (datetime.today() - timedelta(days=1)).replace(tzinfo=utc),
+                'intro_video': course_details.intro_video
+            }
```

Comment:
We need to add start_date because otherwise this method clears the start date and next time a user tries to view the course, the page crashes as it tries to get the start date to show in the top menu. Also, intro_video needs to be added because this method directly updates this field in the course settings without checking if it exists in the passed dict hence leading to a crash.

Reply:
@could you please use the `start_data` of course as you are using the `intro_video`?

Reply:
@ziafazal it maybe that `start_date` is in the future and when I set the `end_date` in the past, it raises the error that `start_date` can not be greater than end_date. that's the reason I did this.

Reply:
turns out it doesnt raise an error. the check is only in studio. I have set it as told.